### ISSUE TITLE
Memory error on Hello World program

### ISSUE THREAD

**Issue by AshvinVaidyanathan** [riscv/riscv-isa-sim#1163]
I try to run the Hello World example using the  `/path/to/spike /path/to/pk hello`  command, but I get this error.

```
Access exception occurred while host was accessing memory on behalf of target (tohost = 0x1008000ea98):
Memory address 0x1008000ea98 is invalid
```

**Comment by scottj97** [comment]
The problem is that your `hello` program uses addresses where Spike doesn't have any memory.

Two options: either change the way `hello` is compiled to use the default memory range supported by Spike (0x80000000-0xffffffff) or use the `-m` option to Spike to set up memory where you need it.

**Comment by AshvinVaidyanathan** [comment]
> either change the way hello is compiled to use the default memory range supported by Spike (0x80000000-0xffffffff)

I know this is beyond the scope of this repo but could you elaborate a bit on how I could do that? I'm pretty new to this stuff and haven't used the compiler beyond the simple compiling and disassembling.

**Comment by scottj97** [comment]
The easiest thing is to look at the disassembly of hello and see what addresses are used. Then use `-m` to put memories in those places in Spike.

To change `hello`'s memory map would (I guess?) require a custom linker script for the compiler toolchain.

**Comment by AshvinVaidyanathan** [comment]
Hi @scottj97, I decided to reinstall spike and pk before I went ahead with your suggestion because I had installed the two a long time ago. Now it works perfectly fine. What could be the reason a clean install fixed it or rather why was the issue popping up in the first place. I know it's hard to know without actually seeing what was being done on my side, but I am just looking for any educated guesses. 

Thanks a lot for your help so far.
